misc/sieve_of_eratosthenes.c

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file
+ * @brief Get list of prime numbers using Sieve of Eratosthenes
```

Comment:
Provide a Wikipedia link in Markdown format (if possible).
If there's no Wikipedia link, provide another web source for algorithm explanation. 🙂